src/coreclr/src/vm/dwreport.cpp

```diff
@@ -189,14 +189,17 @@ HRESULT DwGetFileVersionInfo(
     CONTRACTL
     {
         NOTHROW;
-        GC_NOTRIGGER;
+        GC_TRIGGERS;
         MODE_ANY;
     }
     CONTRACTL_END;
 
     major = minor = build = revision = 0;
     ULARGE_INTEGER appVersion = { 0, 0 };
 
+    // Preempt to let GC suspend
```

Comment:
It would be better to switch to pre-emptive even earlier on the stack. The more code runs in preeemptive mode, the better.

Reply:
can try to switch to pre-emptive in `GetBucketParametersForManagedException`. It does have an `OBJECTREF pThrowable` which is `GCPROTECTED` earlier.. so assume that should be ok?

Reply:
`GetBucketParametersForManagedException` maybe too much trouble. `BaseBucketParamsManager::GetAppVersion` and similar functions may work out better.